p2pserver/message/types/consensus_payload.go

```diff
@@ -90,49 +92,75 @@ func (this *ConsensusPayload) Serialize(w io.Writer) error {
 	buf := keypair.SerializePublicKey(this.Owner)
 	err = serialization.WriteVarBytes(w, buf)
 	if err != nil {
-		return err
+		return errors.NewDetailErr(err, errors.ErrNetPackFail, fmt.Sprintf("write publickey error. publickey buf:%v", buf))
 	}
 
 	err = serialization.WriteVarBytes(w, this.Signature)
 	if err != nil {
-		return err
+		return errors.NewDetailErr(err, errors.ErrNetPackFail, fmt.Sprintf("write Signature error. Signature:%v", this.Signature))
 	}
 
-	return err
+	return nil
 }
 
 //Deserialize message payload
 func (this *ConsensusPayload) Deserialize(r io.Reader) error {
 	err := this.DeserializeUnsigned(r)
-
+	if err != nil {
+		return err
+	}
 	buf, err := serialization.ReadVarBytes(r)
 	if err != nil {
-		log.Warn("Consensus item Owner deserialize failed, " + err.Error())
-		return errors.New("Consensus item Owner deserialize failed.")
+
+		return errors.NewDetailErr(err, errors.ErrNetUnPackFail, "read buf error")
 	}
 	this.Owner, err = keypair.DeserializePublicKey(buf)
 	if err != nil {
-		log.Warn("Consensus item Owner deserialize failed, " + err.Error())
-		return errors.New("Consensus item Owner deserialize failed.")
+
+		return errors.NewDetailErr(err, errors.ErrNetUnPackFail, "deserialize publickey error")
 	}
 
 	this.Signature, err = serialization.ReadVarBytes(r)
 	if err != nil {
-		log.Warn("Consensus item Signature deserialize failed, " + err.Error())
-		return errors.New("Consensus item Signature deserialize failed.")
+
+		return errors.NewDetailErr(err, errors.ErrNetUnPackFail, "read Signature error")
 	}
 
 	return err
 }
 
 //Serialize message payload
 func (this *ConsensusPayload) SerializeUnsigned(w io.Writer) error {
-	serialization.WriteUint32(w, this.Version)
+	err := serialization.WriteUint32(w, this.Version)
+	if err != nil {
+
+		return errors.NewDetailErr(err, errors.ErrNetPackFail, fmt.Sprintf("write error. version:%v", this.Version))
+	}
 	this.PrevHash.Serialize(w)
```

Comment:
err = this.PrevHash.Serialize(w) ???

same comments for followings.

Reply:
Add more check in latest code